posthog/exception_capture.py

```diff
@@ -1,24 +1,46 @@
 import logging
 import sys
 import threading
-from typing import TYPE_CHECKING
+from enum import Enum
+from typing import TYPE_CHECKING, List, Optional
 
-from posthog.exception_utils import exceptions_from_error_tuple
+from posthog.exception_utils import exceptions_from_error_tuple, handle_in_app
+from posthog.utils import remove_trailing_slash
 
 if TYPE_CHECKING:
     from posthog.client import Client
 
 
+class Integrations(str, Enum):
+    Django = "django"
+
+
+DEFAULT_DISTINCT_ID = "python-exceptions"
+
+
 class ExceptionCapture:
     # TODO: Add client side rate limiting to prevent spamming the server with exceptions
 
     log = logging.getLogger("posthog")
 
-    def __init__(self, client: "Client"):
+    def __init__(self, client: "Client", integrations: Optional[List[Integrations]] = None):
         self.client = client
         self.original_excepthook = sys.excepthook
         sys.excepthook = self.exception_handler
         threading.excepthook = self.thread_exception_handler
+        self.enabled_integrations = []
+
+        for integration in integrations or []:
+            # TODO: Maybe find a better way of enabling integrations
+            # This is very annoying currently if we had to add any configuration per integration
+            if integration == Integrations.Django:
+                try:
+                    from posthog.exception_integrations.django import DjangoIntegration
+
+                    enabled_integration = DjangoIntegration(self.exception_receiver)
+                    self.enabled_integrations.append(enabled_integration)
+                except Exception as e:
+                    self.log.exception(f"Failed to enable Django integration: {e}")
```

Comment:
it'd be cool if the integration could configure itself.

```
integrations = [DjangoIntegration]
init(client, integrations)
```

The `Integration` interface has a `install|uninstall` method that is called after the SDK is init.
the `install` method in the `DjangoIntegration` class does all the config.
if the SDK close is called, we also call all `uninstall` methods from all `integrations`
an example https://github.com/PostHog/posthog-android/blob/bb0f3d30cc3d9819000b43c281296a30483841fa/posthog-android/src/main/java/com/posthog/android/internal/PostHogActivityLifecycleCallbackIntegration.kt#L73-L77

Reply:
nice, will do

Reply:
maybe integrations can be also a property in the `client`

```
client.integrations = [DjangoIntegration]
init(client)
```

Reply:
was a bit reluctant to spend more time on this interface, given i'm not yet sure how most integrations will look like. If they need to be initialised post client initialisation / need some config options from users to happen first, would make sense. We can make breaking changes here though as long as its in alpha, so not too worried.